src/numerics/petsc_vector.C

```diff
@@ -1367,8 +1372,12 @@ void PetscVector<T>::_get_array() const
             _first = static_cast<numeric_index_type>(petsc_first);
             _last = static_cast<numeric_index_type>(petsc_last);
           }
-
+#ifdef LIBMESH_HAVE_CXX11_THREAD
+	  std::atomic_thread_fence(std::memory_order_release);
```

Comment:
I think this line needs to be indented one more space... it's not lined up quite right.